master/buildbot/scripts/base.py

```diff
@@ -58,6 +58,7 @@ def isBuildBotRunning(basedir, quiet):
                 printMessage(
                         message="Removing twistd.pid, file has pid {} but buildbot is not running".format(pid),
                         quiet=quiet)
+                f.close()
```

Comment:
`with` enclosure already closes files when it ends. Instead of doing this, you could just move the `os.unlink` file outside of the `with` context.